File: java/src/main/java/com/boo/java/MainActivity.java
package com.boo.java;

import android.os.Bundle;
import android.util.Log;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.boo.java.utils.DefaultUtils;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "onStart: ");
    }

    /**
     * LeetCode马戏团人塔
     */
    private void test1() {
        int[] height = {65, 75, 56, 75, 60, 68};
        int[] weight = {100, 150, 90, 190, 95, 110};
        int a = DefaultUtils.bestSeqAtIndex(height, weight);
        Log.d(TAG, "test1: " + a);
    }

    /**
     * 1.两个线程同时运行；
     * 2.两个线程同时运，int a = 0 是公共代码；
     * 3.synchronized关键字修饰；
     */
    private void test2() {
        new Thread(this::test2Test).start();
        new Thread(this::test2Test).start();
    }

    /**
     * i++、++i区别
     */
    synchronized private void test2Test() {
        int a = 0;
        int b = 0;
        int c = 0;
        int d;
        int e = 0;
        int f;
        for (int i = 0; i < 10; i++) {
            a++;
            ++b;
            d = c++;
            f = ++e;
            Log.d(TAG, "test2Test: a=" + a + ",b=" + b + ",d=" + d + ",f=" + f);
        }
    }
}

File: app/src/main/java/com/boo/app/MainActivity.kt
package com.boo.app

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.boo.app.music.MusicListActivity
import com.boo.app.music.MusicPlayActivity
import com.boo.app.utils.SpfUtils
import com.boo.app.utils.UiUtils

class MainActivity : AppCompatActivity() {
    companion object {
        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "onCreate: ")
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        setDefaultUI()
    }

    /**
     * 根据UI切换初始列表界面
     */
    fun setDefaultUI() {
        val intent = Intent()
        val uiName: String = SpfUtils.getString(UiUtils.SAVE_UI, UiUtils.UI_A)
        Log.d(TAG, "setDefaultUI: uiName = $uiName")
        when (uiName) {
            UiUtils.UI_A -> intent.setClass(this, MusicListActivity::class.java)
            else -> intent.setClass(this, MusicPlayActivity::class.java)
        }
        startActivity(intent)
    }
}